concrete/controllers/panel/dashboard.php

```diff
@@ -82,4 +84,119 @@ public function view()
         $this->set('ui', $ui);
         $this->set('tab', $tab);
     }
+
+    //public function checkForMenu($c) 
+    //{ 
+    //}
+    public function checkForMenu($c) 
+    { 
+      //return (string) $this->renderActiveSubPanels($c) !== ''; // fails to pass triple equal sign checking on element where this function is called.
+      return !($result = $this->renderActiveSubPanels($c)) || $result == '' ? null : 1; 
+    }
+
+    public function renderActiveSubPanels($c)
+    {
+        $nh = Core::make('helper/navigation');
+        $trail = $nh->getTrailToCollection($c);
+        if (
+            $c->getCollectionPath() == '/dashboard' ||
+            $c->getCollectionPath() == '/dashboard/system'
+           ) {
+            $emulatePageID = $c->getCollectionID();
+            $emulate = true;
+        }
+        if (
+            $c->getCollectionPath() == '/account/edit_profile' ||
+            $c->getCollectionPath() == '/account/messages'
+           ) {
+            $emulatePageID = $c->getCollectionParentID();
+            $emulate = true;
+        }
+        if ($emulate === true) {
+            $emulatePage = Page::getByID($emulatePageID);
+            $trail = [$emulatePage, 0];
+        }
+        if (is_array($trail)) {
+            array_pop($trail);
+            $trail = array_reverse($trail);
+            ob_start();
+            for ($i = 0; $i < count($trail); ++$i) {
+                $page = $trail[$i];
+                $label = $page->getCollectionName();
+                $cID = $page->getCollectionID();
+                $parentID = $page->getCollectionParentID();
+                $parent = Page::getByID($parentID);
+                $parentTitle = $parent->getCollectionName();
+                $parentParentID = $parent->getCollectionParentID();
+                $parentParent = Page::getByID($parentParentID);
+                $parentParentTitle = $parentParent->getCollectionName();
+                $parentMenu = new PageList();
+                $parentMenu->filterByExcludeNav(false);
```

Comment:
Doing these page lists inside a for loop seems like it's going to very negatively impact performance.

Reply:
I'll see what I can do to avoid using PageList.  Maybe something like the way the old dashboard panel would render submenus whenever an item was drilled into by using a walk function?

Reply:
@aembler At most the for loop is only going to loop twice, three at the most.  It loops into the current submenu.  So if your inside '/dashboard/users/groups/message', the loop is going to create a pagelist for 'group', and then one for 'users'.  Most of the time there is only one sub panel loaded, causing the for loop to only iterate once.

Reply:
@aembler What do you think would be a better way?